docs/components/file-upload.md

```diff
@@ -0,0 +1,310 @@
+    </KFileUpload>
+  </template>
+</KCard>
+
+```html
+<KFileUpload 
+  type="image"
+  :removable=false
+  label="Upload File"
+  placeholder="No cancel button!" 
+  icon="kong"
+/>
+```
+
+### accept
+
+Please refer to [mdn web docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#accept) that defines the file types the file input should accept. 
+
+### buttonAppearance
+
+Use this prop to customize the trigger **KButton** [appearance](/components/button.html#appearance). This button only shows when prop `type` is `file`. Default vaue is set to `primary`.
+
+<KCard>
+  <template v-slot:body>
+    <KFileUpload label="Upload File" :label-attributes="{ help: `Accepted file types: ${acceptedFileType}` }" buttonAppearance="creation" :accept="acceptedFileType" />
+  </template>
+</KCard>
+
+```html
+<KFileUpload label="Upload File" buttonAppearance="creation" />
+```
+
+### buttonText
+
+This is the text that will be displayed on the button that triggers the click on KInput. Button text can only be changed when prop `type` is `file`.
+
+<KCard>
+  <template v-slot:body>
+    <KFileUpload label="Upload File" :label-attributes="{ help: `Accepted file types: ${acceptedFileType}` }" buttonAppearance="danger" buttonText="Click me" :accept="acceptedFileType" />
+  </template>
+</KCard>
+
+```html
+<KFileUpload label="Upload File" buttonAppearance="danger" buttonText="Click me" />
+```
+
+### icon
+
+Specify an icon to display to the left of the `placeholder text` if prop `type` is `image`.
+
+### iconSize
+
+The size of the `icon` being displayed (default is `26`).
+
+### iconColor
+
+The color of the `icon` being displayed.
+
+<KCard class="mt-6">
+  <template v-slot:body>
+    <KFileUpload type="image" label="Upload File" :label-attributes="{ help: `Accepted file types: ${acceptedImageType}` }" :accept="acceptedImageType" class="image-with-label" placeholder="Customized icon, iconColor & iconSize!" icon="immunity" iconColor="gold" iconSize="30" />
+  </template>
+</KCard>
+
+```html
+<KFileUpload 
+  type="image" icon="immunity" iconColor="gold" iconSize="30" 
+  placeholder="You can change the text here!"
+/>
+```
+### maxFileSize
+
+Use this prop to customize the maximize size of file that can be uploaded. Default value for `image` upload is `1mb (1000000 bytes)` and for `file` is `5.24mb (5242880 bytes)`.
+
+## Events
+
+The events below will fire whenever:
+
+- a file is added `file-added`
+- a file is removed `file-removed`
+- there is a file upload error `error`
+
+<KCard>
+  <template v-slot:body>
+    <KFileUpload label="Upload File" :label-attributes="{ help: `Accepted file types: ${acceptedFileType}` }" :accept="acceptedFileType" @file-added="file => printData(file)" @file-removed="() => { fileData = '' }" />
+  </template>
+</KCard>
```

Comment:
You need to put the example code below this block so people can see how it's being used in the docs.